source/common/upstream/cluster_manager_impl.cc

```diff
@@ -281,23 +305,31 @@ void ClusterManagerImpl::ThreadLocalClusterManagerImpl::shutdown() {
 ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::ClusterEntry(
     ThreadLocalClusterManagerImpl& parent, const Cluster& cluster, Runtime::Loader& runtime,
     Runtime::RandomGenerator& random, Stats::Store& stats_store, Event::Dispatcher& dispatcher,
-    const std::string& local_zone_name, const std::string& local_address)
+    const std::string& local_zone_name, const std::string& local_address,
+    const Optional<std::string>& local_cluster_name)
```

Comment:
Just pass the resolved local cluster into here or nullptr. You already do the resolution in ThreadLocalClusterManagerImpl() constructor